### ISSUE TITLE
Refresh ack timeout

### ISSUE THREAD

**Issue by TroyKomodo** [rabbitmq/rabbitmq-server#4029]
Hi, I am unsure if this is a thing, but...
If we have a consumed queue, and then we consume a task on the queue by default the ack timeout is 30min if the task is not ack'ed within that 30min window the connection is closed and then requeued. I know I can modify the global timeout on the RMQ instance but that seems like a strange way to fix this problem. Ideally there should be a way to refresh ownership of a task as to not timeout. This is likely the wrong repo I am making this issue on but I am also unsure where I should suggest a feature.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
As [documented in the Consumers guide](https://rabbitmq.com/consumers.html#acknowledgement-timeout).

**Comment by lukebakken** [comment]
Related to #2990

For now you will have to adjust the global consumer timeout if you need more time to complete the task.

RabbitMQ can't release resources for a queued item until the consumer acks it. With enough un-acked items We had enough experience with support cases arising from abusive consumers that the timeout was implemented.

**Comment by TroyKomodo** [comment]
I think you are correct, there should be some progress metric that would make the most sense, although might require some restructuring. The problem came up with transcoding I used RabbitMQ as a queue to issue transcoding tasks for video. The transcoder picks up the task and then acks when the transcoding job is complete. Sometimes, if the videos are several hours the job would take longer than 30min. I have since changed the timeout to 12h but I thought maybe there should be a feature to resolve this.
